packages/ts-codegen/src/context/index.ts

```diff
@@ -127,12 +140,20 @@ export class TsCodegenContext {
   }
 
   getAllImportDeclarations(): ts.ImportDeclaration[] {
-    return this.imports.map(({ defs, from }) =>
-      createImportDeclaration(
-        defs.map(({ typeName }) => addModelSuffix(typeName)),
-        from,
-      ),
-    );
+    return this.imports
+      .map(({ defs, from }) => {
+        const filteredDefs = defs.filter(({ typeName }) =>
+          this.entitiesToImport.has(typeName),
+        );
+        if (!filteredDefs.length) {
+          return;
+        }
+        return createImportDeclaration(
+          filteredDefs.map(({ typeName }) => addModelSuffix(typeName)),
+          from,
+        );
+      })
+      .filter(Boolean) as ts.ImportDeclaration[];
```

Comment:
When you do an assertion, you loose benefits. I'm guessing it's the `map` that is loosing the precision? It would be better to pass this as a type parameter to map.

Reply:
Created example comment above, but unsure from the top of my head if that's the right parameter.